### PR TITLE
Add tests for compare op

### PR DESCRIPTION
**Pull request by mmanzoorTT**
* Equal
* Not equal
* greater than
* greater or equal to
* less than
* less or equal to

### CODE REVIEW

File: tests/TTIR/test_compare.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+# The following tests explicitly convert data type of comparison operation
+# output for the verification purposes.
+
+
+def test_equal():
+    def module_equal(a, b):
+        c = a == b
+        return jax.lax.convert_element_type(c, jnp.float32)
+
+    verify_module(module_equal, [(64, 64), (64, 64)], dtype=jnp.bfloat16)
```
Comment (mrakitaTT) [inline tests/TTIR/test_compare.py:25]: Can you please add `parametrize` decorators to each test which take input shapes as parameter, like this:
```
@pytest.mark.parametrize("input_shapes", [[(64, 64), (64, 64)]])
def test_equal():
...
    verify_module(module_equal, input_shapes, dtype=jnp.bfloat16)
```

**Review by mrakitaTT — APPROVED** [review]

File: tests/TTIR/test_compare.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import jax.numpy as jnp
+import numpy
+
+from infrastructure import verify_module
+
+# Note: TTNN does not support boolean data type, so bfloat16 is used instead.
+# Hence the output of comparison operation is bflaot16. JAX can not perform any
+# computation due to mismatch in output data type (in testing infrastructure).
+# The following tests explicitly convert data type of comparison operation
+# output for the verification purposes.
```
[the remark this answers is not in the document]
Comment (mmanzoorTT) [reply tests/TTIR/test_compare.py:17]: Thanks Marko. I have added a comment as suggested.

File: tests/TTIR/test_compare.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+def test_equal():
+    def module_equal(a, b):
+        c = a == b
+        return jax.lax.convert_element_type(c, jnp.float32)
+
+    verify_module(module_equal, [(64, 64), (64, 64)], dtype=jnp.bfloat16)
+
+
+def test_notEqual():
+    def module_notEqual(a, b):
```
[the remark this answers is not in the document]
Comment (mmanzoorTT) [reply tests/TTIR/test_compare.py]: Thanks. names modified.

File: tests/TTIR/test_compare.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    verify_module(module_lessThan, [(64, 64), (64, 64)], dtype=jnp.bfloat16)
+
+
+def test_lessEqual():
+    def module_lessEqual(a, b):
+        c = a <= b
+        return jax.lax.convert_element_type(c, jnp.float32)
+
+    verify_module(module_lessEqual, [(64, 64), (64, 64)], dtype=jnp.bfloat16)
```
[the remark this answers is not in the document]
Comment (mmanzoorTT) [reply tests/TTIR/test_compare.py]: It already has a new line at the end. GH doesn't show the ending new line.

**Pull request merged**